pkg/cmd/roachtest/test_impl.go

```diff
@@ -487,25 +487,42 @@ func (t *testImpl) failureMsg() string {
 	return b.String()
 }
 
-// failureContainsError returns true if any of the errors in a given failure
-// matches the reference error
-func failureContainsError(f failure, refError error) bool {
-	for _, err := range f.errors {
-		if errors.Is(err, refError) {
+// failuresContainError returns true if any of the errors in any of
+// the given failures matches the reference error.
+func failuresContainsError(failures []failure, refError error) bool {
+	for _, f := range failures {
+		for _, err := range f.errors {
+			if errors.Is(err, refError) {
+				return true
+			}
+		}
+
+		if errors.Is(f.squashedErr, refError) {
 			return true
 		}
 	}
-	return errors.Is(f.squashedErr, refError)
+
+	return false
 }
 
-// failuresContainsError returns true if any of the failures contains the reference error
-func failuresContainsError(failures []failure, refError error) bool {
+// failuresSpecifyOwner checks if any of the errors in any of the
+// given failures is a failure that is associated with issue
+// ownership. If such an error is found, it is returned.
```

Comment:
Nit: "...it is returned; otherwise, nil is returned."

Reply:
Fixed.